File: programmers/DFS_BFS/단어 변환.js
function solution(begin, target, words) {
  const n = words.length;
  visit = Array(n).fill(false);
  q = [[begin, 0]];
  while (q.length) {
    const [word, cnt] = q.shift();
    if (word === target) return cnt;
    for (let i = 0; i < n; i++) {
      if (!visit[i]) {
        let diff_cnt = 0;
        for (let j = 0; j < word.length; j++) {
          if (word[j] !== words[i][j]) {
            diff_cnt++;
          }
          if (diff_cnt > 1) break;
        }
        if (diff_cnt === 1) {
          q.push([words[i], cnt + 1]);
          visit[i] = true;
        }
      }
    }
  }
  return 0;
}

File: programmers/DFS_BFS/단어 변환.py
def solution(begin, target, words):
    n = len(words)
    q = [[begin, 0]]
    visit = [0] * n
    while q:
        word, cnt = q.pop(0)
        if word == target:
            return cnt
        for i in range(n):
            if not visit[i]:
                diff_cnt = 0
                for j in range(len(word)):
                    if word[j] != words[i][j]:
                        diff_cnt += 1
                    if diff_cnt > 1:
                        break
                if diff_cnt == 1:
                    q.append([words[i], cnt + 1])
                    visit[i] = 1
    return 0
